### PR TITLE
feat: create compare date filter

### PR DESCRIPTION
**Pull request by JeevaRamu0104**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
create date compare filter component
image

https://github.com/user-attachments/assets/e8f000be-b955-4d66-ab72-c114f71209f5

## Where to test it?

- [ ] INTEG
- [ ] SANDBOX
- [ ] PROD

## Checklist

- [ ] I ran `npm run re:build`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by kanikabansal-juspay — DISMISSED** [review]

File: src/screens/NewAnalytics/NewAnalyticsContainerUtils.res [hunk 1]
```diff
@@ -17,14 +17,39 @@ let getPageFromIndex = index => {
   | 1 | _ => NewAnalyticsPayment
   }
 }
+let renderValueInp = () => (_fieldsArray: array<ReactFinalForm.fieldRenderProps>) => {
+  React.null
+}
```
Comment (Riddhiagrawal001) [inline src/screens/NewAnalytics/NewAnalyticsContainerUtils.res:22]: Do we need this ?

File: src/components/DateRangeCompareFields.res [hunk 2]
```diff
@@ -0,0 +1,807 @@
...
+    let datevalue = Js.Date.makeWithYMD(
+      ~year=Js.Float.fromString(date[0]->Option.getOr("")),
+      ~month=Js.Float.fromString(
+        String.make(Js.Float.fromString(date[1]->Option.getOr("")) -. 1.0),
+      ),
+      ~date=Js.Float.fromString(date[2]->Option.getOr("")),
+      (),
+    )
+    datevalue
+  }
```
Comment (Riddhiagrawal001) [inline src/components/DateRangeCompareFields.res:37]: Why we are having function inside a function?

File: src/components/DateRangeCompareFields.res [hunk 3]
```diff
@@ -0,0 +1,807 @@
...
+      disablePastDates,
+      todayDate,
+      todayTime,
+    )
+
+    let startDate = getFormattedDate(`${stDate}T${stTime}Z`, formatDateTime)
+    let endDate = getFormattedDate(`${enDate}T${enTime}Z`, formatDateTime)
+    let handleClick = _value => {
+      onClick(value, disableFutureDates)
+    }
```
Comment (Riddhiagrawal001) [inline src/components/DateRangeCompareFields.res:83]: If value is not needed can we remove the parameter?

File: src/components/DateRangeCompareFields.res [hunk 4]
```diff
@@ -0,0 +1,807 @@
...
+    ~compareWithEndTime: string,
+  ) => {
+    open DateRangeHelper
+    open DateRangeUtils
+    open LogicUtils
+    let (isCustomSelected, setIsCustomSelected) = React.useState(_ =>
+      predefinedDays->Array.length === 0
+    )
+
+    let (_showOption, setShowOption) = React.useState(_ => false)
```
Comment (Riddhiagrawal001) [inline src/components/DateRangeCompareFields.res:147]: If we are not using the state value , then do we need this state ?

File: src/screens/NewAnalytics/NewAnalyticsContainerUtils.res (hunk 1 above)
Comment (JeevaRamu0104) [reply src/screens/NewAnalytics/NewAnalyticsContainerUtils.res:22]: yes we require this.

**Review by Riddhiagrawal001 — APPROVED** [review]

**Review by kanikabansal-juspay — APPROVED** [review]

**Pull request merged**